### PR TITLE
[cli] Add Task Scheduling Command and Refactor Wallet Context API

### PR DESCRIPTION
**Pull request by jolestar**
## Summary

This PR introduces several enhancements and new features:

1. Scheduled Task Command: Implements the rooch task schedule command to run scheduled tasks. This includes checking for OnChainEvent and executing transactions to consume these events.
2. Wallet Context Refactor: Refactors the wallet context, focusing on password handling and the sign API to simplify the logic and improve usability.

### CODE REVIEW

File: crates/rooch/src/commands/move_cli/commands/run_function.rs [hunk 1]
```diff
@@ -153,52 +153,28 @@ impl CommandAction<ExecuteTransactionResponseView> for RunFunction {
...
+                let tx_execution_result = context.sign_and_execute(sender, tx_data.clone()).await?;
+
+                if self.gas_profile {
+                    //TODO FIXME we should use the state_root from previous tx
+                    let state_root = tx_execution_result
+                        .execution_info
+                        .state_root
+                        .0
+                        .as_bytes()
+                        .to_vec();
```
Comment (jolestar) [inline crates/rooch/src/commands/move_cli/commands/run_function.rs:168]: @steelgeek091 We use a wrong state_root to execute_tx locally.

Comment (steelgeek091) [reply crates/rooch/src/commands/move_cli/commands/run_function.rs:168]: We getting the last Tx order which has processed by sequencer then use this Tx order to query previous Tx?

File: crates/rooch/src/commands/task/commands/schedule.rs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    stop_on_runner_error: bool,
+
+    /// Stop the schedule task after executed times
+    #[clap(long)]
+    stop_after_executed_times: Option<u64>,
+}
+
+#[async_trait]
+impl CommandAction<String> for ScheduleCommand {
+    async fn execute(self) -> RoochResult<String> {
```
Comment (baichuan3) [inline crates/rooch/src/commands/task/commands/schedule.rs:48]: 目前实现的是单节点下的Schedule执行，多节点后再考虑去中心化的Schedule？

**Review by baichuan3 — APPROVED** [review]

File: crates/rooch/src/commands/task/commands/schedule.rs (hunk 2 above)
Comment (jolestar) [reply crates/rooch/src/commands/task/commands/schedule.rs:48]: Schedule 的应用场景就是单机模式的定时任务。一般应用自己运行一个就行。去中心化的任务系统需要新的协议来搞。

File: crates/rooch/src/commands/move_cli/commands/run_function.rs (hunk 1 above)
Comment (jolestar) [reply crates/rooch/src/commands/move_cli/commands/run_function.rs:168]: If the tx has been executed and you use the state_root to execute it again, the ObjectID and Resource may conflict. Maybe the gas profile mode should only execute the tx off-chain.

**Pull request merged**